File: testdata/precision-suite-swift/clean_parse_id.swift
// Clean: numeric coercion sanitizes the tainted value before it reaches a sink.
// `Int(raw)` returns nil on any non-numeric input, so a path/command
// metacharacter can never survive — the value handed onward is a validated
// integer. A correct scanner emits nothing; any finding here is a false positive.
import Foundation

func openRecord() -> String {
    let raw = CommandLine.arguments[1]
    // Int(...) coercion strips every injection metacharacter (nil otherwise).
    guard let n = Int(raw) else { return "" }
    let path = "/var/records/\(n).txt"
    return (try? String(contentsOfFile: path, encoding: .utf8)) ?? ""
}

File: testdata/precision-suite-dart/clean_parse_id.dart
// Clean: an untrusted value is coerced through `int.parse` before it reaches a
// command. Numeric coercion strips every shell metacharacter (and throws on
// non-numeric input), so the value handed to `Process.run` can only ever be a
// number — no injection is possible. A correct scanner emits nothing; any finding
// here is a false positive. nox's `int.parse` sanitizer neutralizes the injection
// classes on the path.
import 'dart:io';

Future<void> runReport() async {
  final raw = Platform.environment['COUNT'] ?? '0';
  final count = int.parse(raw);
  await Process.run('generate-report', ['--count', '$count']);
}
